bin/resolve.js

```diff
@@ -0,0 +1,291 @@
+              json: { names : [ nameToUse ] },
+            });
+
+            const matches = body['results'].map(result => result['matches']).reduce((acc, curr) => acc.concat(curr), []);
+
+            const ottNames = matches.filter(match => match).map(match => match['taxon']['name']).filter(name => name);
+            const ottIds = matches.filter(match => match).map(match => match['taxon']['ott_id']).filter(ott_id => ott_id);
+
+            if (ottIds.length > 1) debug(`     - Taxon name ${nameComplete} resolved to multiple OTT Ids: ${ottIds.join(', ')}.`)
+
+            const result = {};
+            result[nameComplete] = ottIds;
+            return result;
+          }
+        }
+      }
+
+      debug(`   - Internal specifiers:`);
+      const internalOTTs = wrappedPhyloref.internalSpecifiers.map(specifierToOTLId);
+      const internalOTTids = lodash.flattenDeep(internalOTTs.map(ott => lodash.head(lodash.values(ott))));
+      // console.log(internalOTTids);
+
+      debug(`   - External specifiers:`);
+      const externalOTTs = wrappedPhyloref.externalSpecifiers.map(specifierToOTLId);
+      const externalOTTids = lodash.flattenDeep(externalOTTs.map(ott => lodash.head(lodash.values(ott))));
+      // console.log(externalOTTids);
+
+      if (internalOTTids.filter(x => x === undefined).length > 0) {
+        debug('Not all internal specifiers could be resolved to OTT Ids, skipping.');
+        return {
+          filename,
+          phyloref,
+          error: 'internal_specifiers_missing',
+        };
+      } else if (externalOTTids.filter(x => x === undefined).length > 0) {
+        debug('Not all internal specifiers could be resolved to OTT Ids, skipping.');
+        return {
+          filename,
+          phyloref,
+          error: 'external_specifiers_missing',
+        };
+      } else if (internalOTTids.length == 0) {
+        debug('No internal specifiers found, skipping.');
+        return {
+          filename,
+          phyloref,
+          error: 'no_internal_specifiers',
+        };
+      } else {
+        // debug('Request: ', { node_ids: internalOTTids, excluded_node_ids: externalOTTids });
+        const result = retus("https://api.opentreeoflife.org/v3/tree_of_life/mrca ", {
+          throwHttpErrors: false,
+          method: 'post',
+          json: { node_ids: internalOTTids.map(id => "ott" + id), excluded_node_ids: externalOTTids.map(id => "ott" + id) },
+          responseType: 'text',
+        });
+
+        // There are two types of responses we might get:
+        //  - If we have external specifiers, we'll get a 'node_ids', where the first one
+        //    is the node-based name and the last one is the branch-based name, and a 'synth_id'.
+        //  - If we have only internal specifiers, we'll get a 'mrca' with a 'node_id' (as well as
+        //    'supported_by' and 'unique_name') and a 'synth_id'.
+
+        if (result.statusCode == 404 || result.statusCode == 400) {
```

Comment:
It doesn't seem like you're testing whether the HTTP status code is indeed 200 (or 2xx) if it's not 404 or 400. Is that true, and if so, shouldn't that be tested?

Reply:
Fixed in 91c86ca.